src/bundle/Resources/public/js/CKEditor/anchor/anchor-editing.js

```diff
@@ -17,6 +17,57 @@ class IbexaAnchorEditing extends Plugin {
                 key: 'id',
             },
         });
+
+        this.editor.conversion.for('dataDowncast').add((dispatcher) => {
+            dispatcher.on('attribute:anchor:listItem', (event, data, conversionApi) => {
+                if (data.attributeKey !== 'anchor' || data.attributeNewValue === '') {
```

Comment:
I'm not sure how ckeditor works here, but do you need to check for emptiness of attributeNewValue? As far as I checked, there's form error when you try so save anchor with empty value.
Also, to be sure - when you remove anchor, isn't this callback run with attributes of anchor and empty value?